src/Storefront/Resources/app/storefront/webpack.config.js

```diff
@@ -321,11 +321,19 @@ const pluginConfigs = pluginEntries.map((plugin) => {
     }
 
     if (isHotMode) {
-        if (plugin.isTheme && plugin.name === themeFiles.technicalName) {
+        const scriptAssetNames = [];
+        themeFiles.script.map((script) => {
+            scriptAssetNames.push(script.assetName);
+        });
```

Comment:
Array map already creates a new array with the result. No need to use it as a forEach(). 

```suggestion
        const scriptAssetNames = themeFiles.script.map(script => script.assetName);
```